test/integration/patients/sidebar/flow-sidebar.js

```diff
@@ -617,4 +617,273 @@ context('flow sidebar', function() {
       .get('[data-permission-region]')
       .should('contain', 'You are not able to change settings on flows.');
   });
+
+  specify('flow with only work:team:manage permission', function() {
+    cy
+      .routesForPatientDashboard()
+      .routeCurrentClinician(fx => {
+        fx.data.relationships.role = { data: { id: '77777' } };
+        fx.data.relationships.team = { data: { id: '11111', type: 'teams' } };
+
+        return fx;
+      })
+      .routeWorkspaceClinicians(fx => {
+        fx.data = _.first(fx.data, 3);
+
+        const teamMemberClinician = _.find(fx.data, { id: '22222' });
+        teamMemberClinician.attributes.name = 'Team Member';
+        teamMemberClinician.relationships.team.data.id = '11111';
+
+        const nonTeamMemberClinician = _.find(fx.data, { id: '33333' });
+        nonTeamMemberClinician.attributes.name = 'Non Team Member';
+        nonTeamMemberClinician.relationships.team.data.id = '22222';
+
+        return fx;
+      })
+      .routeFlow(fx => {
+        fx.data.id = '1';
+        fx.data.attributes.name = 'Owned by current clinician’s team';
+        fx.data.relationships.state = { data: { id: '33333', type: 'states' } };
+        fx.data.relationships.owner = { data: { id: '11111', type: 'teams' } };
+        fx.data.relationships.patient.data.id = '1';
+
+        fx.included.push({
+          id: '1',
+          attributes: {
+            first_name: 'Test',
+            last_name: 'Patient',
+          },
+          type: 'patients',
+        });
+
+        return fx;
+      })
+      .routePatientFlows(fx => {
+        fx.data = _.sample(fx.data, 4);
+
+        fx.data[0].id = '1';
+        fx.data[0].attributes.name = 'Owned by current clinician’s team';
+        fx.data[0].attributes.updated_at = testTsSubtract(1);
+        fx.data[0].relationships.state = { data: { id: '33333' } };
+        fx.data[0].relationships.owner = { data: { id: '11111', type: 'teams' } };
+
+        fx.data[1].id = '2';
+        fx.data[1].attributes.name = 'Owned by another team';
+        fx.data[1].attributes.updated_at = testTsSubtract(2);
+        fx.data[1].relationships.state = { data: { id: '33333' } };
+        fx.data[1].relationships.owner = { data: { id: '22222', type: 'teams' } };
+
+        fx.data[2].id = '3';
+        fx.data[2].attributes.name = 'Owned by team member';
+        fx.data[2].attributes.updated_at = testTsSubtract(3);
+        fx.data[2].relationships.state = { data: { id: '33333' } };
+        fx.data[2].relationships.owner = { data: { id: '22222', type: 'clinicians' } };
+
+        fx.data[3].id = '4';
+        fx.data[3].attributes.name = 'Owned by non team member';
+        fx.data[3].attributes.updated_at = testTsSubtract(4);
+        fx.data[3].relationships.state = { data: { id: '33333' } };
+        fx.data[3].relationships.owner = { data: { id: '33333', type: 'clinicians' } };
```

Comment:
Looking at these, we could probably shorten up the `action-sidebar.js`, `flow-sidebar.js`, and `patient-flow.js` tests by only testing the restrictive effects of the permission.

So in this test we'd only have two flows: `'Owned by another team'` and `'Owned by non team member'`. And make sure those flows can't be edited.

`'Owned by current clinician’s team'` and `'Owned by team member'` test that the user can edit the flow.

Idk not sure... but some of these tests got kind of big, so it'd be nice to shrink them somehow.

Reply:
Yeah it's funny..  I'm not sure on where the consistency is.  Semantically, the default stuff we shouldn't need to test for is the least permissive state, and a "user _with_ a permission" would be tested to see if they can do things, but as is the nature of things, most of our tests are built from full permissions and we're adding tests for restrictions.  

I don't know the best way to do this long term, but it does seem like in an ideal world, the "generic" tests would be with as little permissions as possible, and all other functionality is tested in the context of permissions, but that seems like an overhaul.  Possibly we should be testing the least permissive scenario and the most permissive scenario for functionality, and then beyond that, the most limited tests possible for combinations in between.  

For instance these tests are all checking to make sure that both the state and the owner are available or not in the sidebar.. but we have already tested that those things would be available or not based on the results of the `flow.canEdit` function.. For other permissions that change the result of `canEdit` we should have the most minimal tests possible to determine that result.  We don't need each test of a `canEdit` result to re-test everything that happens if that is `true` or `false`

Reply:
Not sure this is helpful, but the way we test permissions in BE is to create a fixture with the smallest set of permissions possible for a scenario and then testing it. We don't bother testing the whole matrix, because (as above) the most permissive always takes precedence.

Reply:
For a start, I just pushed a commit to refactor the worklist/schedule tests.

In those tests, I stripped down the test to only include the two restrictive scenarios:

1. `Owned by another team`
2. `Owned by non team member`

And then made the test simply verify that those two actions/flows in the list are in a `canEdit = false` state. Everything else in the test was removed.

Is that all that needs to be done to refactor these tests?

*One caveat is that I left the `Owned by current clinician’s team` scenario in the one of the worklist tests so I could test the contents of the owner droplist (where it should only include the user's team and team members). That can be removed later when we create a component test for the owner droplist.*